planner/core/stats.go

```diff
@@ -506,10 +508,8 @@ func (ds *DataSource) accessPathsForConds(conditions []expression.Expression, us
 				logutil.BgLogger().Debug("can not derive statistics of a path", zap.Error(err))
 				continue
 			}
-			// If `AccessConds` is empty, we ignore the access path.
-			// If the path contains a full range, ignore it also. This can happen when `AccessConds` is constant true, and
-			// it comes from the result of a subquery, so it is not folded.
-			if len(path.AccessConds) == 0 || ranger.HasFullRange(path.Ranges) {
+			// If the path contains a full range, ignore it.
+			if ranger.HasFullRange(path.Ranges) {
 				continue
```

Comment:
Why is this changed too?

Reply:
It is a bug, if the filter is like `(a > 1 and a < 0) or b > 1`, for index a, both `path.AccessConds` and `path.Ranges` are empty, we should generate an `IndexMerge` plan on `TableDual`, not a `TableFullScan` plan.

Reply:
OK.